src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/ContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Autofac;
+using Microsoft.Extensions.Logging;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    public static class ContainerBuilderExtensions
+    {
+        public static void RegisterLogging(this ContainerBuilder builder, ILoggerFactory loggerFactory)
+        {
+            builder
+                .RegisterInstance(loggerFactory)
+                .AsSelf()
+                .As<ILoggerFactory>();
```

Comment:
Do we need `AsSelf()` here?

Reply:
using `IServiceCollection.AddLogging()`, so this was removed